substrate/frame/support/src/storage/bounded_btree_map.rs

```diff
@@ -17,9 +17,10 @@
 
 //! Traits, types and structs to support a bounded BTreeMap.
 
-use crate::storage::StorageDecodeLength;
+use crate::storage::{StorageDecodeLength, StorageDecodeNonDedupLength};
 pub use sp_runtime::BoundedBTreeMap;
 
+impl<K, V, S> StorageDecodeNonDedupLength for BoundedBTreeMap<K, V, S> {}
```

Comment:
I think there could be a blanked impl such that `StorageDecodeNonDedupLength` is implemented for all `StorageDecodeLength`.  
Not sure if we really need it on anything besides the `BTreeSet`.